### PR TITLE
Add beat.version field to all events

### PR DESCRIPTION
**Pull request by ruflin**
Having beat.version will simplify migration events from one version to an other as events can be for example reindex based on the version. This should also simplify debugging as events can be related to a beat version.

* Move version.go to its own package to prevent circular dependencies
* Add dashboard compiling to travis build to make sure it always works
* Update mappings with new field
* Fix local fields from elastic#2689

### CODE REVIEW

File: libbeat/publisher/client.go [hunk 1]
```diff
@@ -76,6 +77,7 @@ func newClient(pub *BeatPublisher) *client {
 		beatMeta: common.MapStr{
 			"name":     pub.name,
 			"hostname": pub.hostname,
+			"version":  version.GetDefaultVersion(),
```
Comment (tsg) [inline libbeat/publisher/client.go]: Is there any way for a community Beat to have their own version? I think we have a way in other places where the version is involved.

**Comment by monicasarbu** [comment]
@ruflin Can you please rebase here, after the #2704 was merged?

File: libbeat/publisher/client.go (hunk 1 above)
Comment (andrewkroh) [inline libbeat/publisher/client.go]: Maybe the version can be injected into the `BeatPublisher` like the name is.

**Comment by ruflin** [comment]
@monicasarbu already done.

File: libbeat/publisher/client.go (hunk 1 above)
Comment (ruflin) [inline libbeat/publisher/client.go]: @tsg good point, either I go the approach suggested by @andrewkroh or introduce a SetVersion to set the final version here: https://github.com/elastic/beats/pull/2700/files#diff-4cc56dbce8e84b1bc2a025380117788eR139

Looking into it. This should be definitively in sync with the beat version itself.

Comment (ruflin) [inline libbeat/publisher/client.go]: @tsg @andrewkroh I rewrote it by passing through the param. Have a look at it again.

**Comment by monicasarbu** [comment]
LGTM

**Comment by monicasarbu** [comment]
I would say it's safe to backport it to 5.0

**Comment by ruflin** [comment]
jenkins, retest it

**Pull request merged**